### PR TITLE
Create campaign modal

### PR DESCRIPTION
**Pull request by MusikAnimal**
Bug: https://phabricator.wikimedia.org/T144864

### CODE REVIEW

File: app/assets/stylesheets/modules/_campaign.styl [hunk 1]
```diff
@@ -46,3 +46,9 @@
 
     .campaign-use-dates
       display: block
+
+.create-campaign-button
+  margin-bottom: 50px
+
+.wizard .right
+  width: 100% !important
```
Comment (ragesoss) [inline app/assets/stylesheets/modules/_campaign.styl]: This seems very dangerous, as `wizard` is used in a lot of places. Can we find a better way to ensure the correct width in this context without `!important`?

File: app/views/campaigns/index.html.haml [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
   .container
     %hr
-    %h1
-      Create New Campaign
-    %div
-      =form_tag do
-        = text_field :campaign, 'title', placeholder: 'Title'
-        = submit_tag 'Create Campaign', class: 'button dark'
-%section
+    - if current_user && Features.open_course_creation?
```
Comment (ragesoss) [inline app/views/campaigns/index.html.haml]: Admins on Wiki Ed dashboard also need to be able to create a campaign, even with open_course_creation disabled.

File: app/views/campaigns/_create_modal.html.haml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                = label(:campaign, :start, t('courses.creator.start_date') + ':')
+                = date_field(:campaign, :start, placeholder: '2016-12-31')
+              .form-group
+                = label(:campaign, :end, t('courses.creator.end_date') + ':')
+                = date_field(:campaign, :end, placeholder: '2016-12-31')
+
+          .column
+            .form-group
+              = label(:campaign, :description, t('campaign.description') + ':')
+              = text_area(:campaign, :description)
```
Comment (ragesoss) [inline app/views/campaigns/_create_modal.html.haml]: This needs a maximum size to guarantee that it won't be more than the database will accept.

File: app/models/campaign.rb [hunk 4]
```diff
@@ -118,6 +124,11 @@ def validate_dates
     end
   end
 
+  def set_slug
+    # Strip everything but letters and digits, and convert spaces to underscores
+    self.slug = title.downcase.gsub(/[^\w0-9 ]/, '').tr(' ', '_') unless self.slug.present?
```
Comment (MusikAnimal) [inline app/models/campaign.rb]: #1062

**Pull request merged**